app/models/proposal.rb

```diff
@@ -45,48 +41,46 @@ def votes_in_tree
     # Rails.cache.fetch("/proposal/#{self.root.id}/votes_in_tree/#{updated_at}", :expires_at => 5.minutes) do
     # Code above seemed to never expire as of Rails 4
     # Proper cache should only cache when votes_in_tree > 100 or so
-    [self.root, self.root.descendants].flatten.map(&:votes_count).sum
+    all_related_proposals.map(&:votes_count).sum
     # end
   end
 
-  # def self_and_descendants    # dupe of all_related_proposals
-  #   [self.root, self.root.descendants].flatten
-  # end
-
   def related_proposals(related_sort_by = 'Most Votes')
```

Comment:
I'm not a fan of this method, mostly because switching the sorting function off a string like this is a bad code smell. These functions are similar enough that this method could be heavily simplified or abstracted. In-place sorting should also be avoided if possible, especially in a situation like this where replacing `sort!` with `sort` wouldn't change the function.